test/acceptance/features/steps/steps.py

```diff
@@ -144,27 +149,39 @@ def db_instance_is_running(context, db_name):
 @when(sbr_is_applied_step)
 def sbr_is_applied(context):
     sbr_yaml = context.text
-    sbr = ServiceBindingRequest()
-    if context.__contains__("nodejs_app"):
-        application = context.nodejs_app
-        context.nodejs_app_original_generation = application.get_observed_generation()
-        context.nodejs_app_original_pod_name = application.get_running_pod_name()
+    namespace = context.namespace
+    sbr = ServiceBindingRequest(namespace.name)
+    if context.__contains__("application"):
+        application = context.application
+        context.application_original_generation = application.get_observed_generation()
+        if context.application_type.__contains__("knative"):
+            context.application_original_rev_name = application.get_rev_name()
+        else:
+            context.application_original_pod_name = application.get_running_pod_name()
     sbr.create(sbr_yaml) | should.be_truthy.desc("Service Binding Request Created")
 
-
 # STEP
+
+
 @then(u'application should be re-deployed')
 def then_application_redeployed(context):
-    application = context.nodejs_app
-    application.get_redeployed_pod_name(context.nodejs_app_original_pod_name) | should_not.be_none.desc(
-        "There is a running pod of the application different from the original one before redeployment.")
+    application = context.application
+    if context.application_type.__contains__("knative"):
+        application.get_redeployed_rev_name(context.application_original_rev_name) | should_not.be_none.desc(
+            "There is a running revision of the application different from the original one before redeployment")
+    else:
+        application.get_redeployed_pod_name(context.application_original_pod_name) | should_not.be_none.desc(
+            "There is a running pod of the application different from the original one before redeployment.")
 
 
 # STEP
 @then(u'application should be connected to the DB "{db_name}"')
 def then_app_is_connected_to_db(context, db_name):
-    application = context.nodejs_app
-    app_db_name = application.get_db_name_from_api()
+    application = context.application
+    if context.application_type.__contains__("knative"):
+        app_db_name = application.get_db_name_from_knative_api(timeout=180)
+    else:
+        app_db_name = application.get_db_name_from_api()
     app_db_name | should.be_equal_to(db_name)
```

Comment:
Instead of doing if/then, how about that both `NodeJSApp` and `QuarkusApplication` implement the same method, e.g. `get_db_name`? In that case, the code would become:

```python
def then_app_is_connected_to_db(context, db_name):
  app_db_name = context.application.get_db_name()
  app_db_name | should.be_equal_to(db_name)
```

Reply:
@pedjak Do you think introducing an interface (say application) with common methods would help in such situations in future test scenarios?

Reply:
not sure if an interface is really required - I mean more about Duck Typing (https://realpython.com/lessons/duck-typing/)

Reply:
I was thinking about future test cases where we might need to add more applications and similar functions might be needed.

Reply:
made to this for now. 
`application = context.application
    app_db_name = application.get_db_name_from_api()
    app_db_name | should.be_equal_to(db_name)` and renamed the function as get_db_name_from_api under QuarkusApplication
Let me know if this fine.